squidDrone/src/com/com_interface.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef SRC_COM_INTERFACE_HPP_
+#define SRC_COM_INTERFACE_HPP_
+
+#ifndef UNIT_TEST
+#include "com_message_buffer.hpp"
+#else
+#include "com_message_buffer_mock.hpp"
+#endif
+#include "com_types.hpp"
+
+#include <array>
+
+namespace com {
+
+  /**
+   * @class ComInterface 
+   * @brief This is a purely virtual class to act as
+   * a facade for the device specific implmentation.
+   * 
+   */
+  class ComInterface {
+    public:
+      /** 
+       * @brief Default constructor may not be used.
+       * 
+       */
+      ComInterface() = delete;
+
+      /** 
+       * @brief Construct a new Com Interface object
+       * 
+       */
+      explicit ComInterface(std::unique_ptr<com::ComMessageBuffer> msg_buf) : msg_buffer_(std::move(msg_buf)){};
+
+      /**
+       * @brief Virtual destructor. 
+       *
+       */
+      virtual ~ComInterface() = default;
+
+      /** 
+       * @brief Get the data packet object
+       * 
+       * @return decltype(array<std::uint8_t, 32>) 
+       */
+      virtual auto get_data_packet() const noexcept -> std::array<std::uint8_t, 32> = 0;
```

Comment:
None of your function names comply with google c++ style guide.

Reply:
Changed function names